### DISCUSSION TITLE
角色、武器信息的材料加个名字

### DISCUSSION THREAD

**Discussion by Arondight** [Arondight/Adachi-BOT#938] [Ideas]
有这么个想法不知道老大哥有没有兴趣未来做一下，图太小看不清，周天也不知道该打啥，如果在哪里能加上“自由”、“诗文”这种文本就好了

![图片](https://user-images.githubusercontent.com/13673702/194592683-509e41f4-4efe-4190-a3b4-1f71c4470e80.png)

图上三行小图标我觉得需要表明的有 ：

1. 第一行的孢子，一般都会提前刷
2. 第二行的天赋书，周本不需要标注，因为没法选材料
3. 第三行的小 boss 材料

**Comment by Arondight** [comment]
有必要的话可以加一个 resources/etc/doc/boss.json 到 CI 里

**Comment by mark9804** [comment]
可以啊，不过要加 boss 的话可能得 url 传一下，为了这一点点信息去新开一个 fetch 时间成本太高了

**Reply by Arondight** [reply]
那干脆把以前那个材料的也 url 传吧，我不知道 fetch 消耗这么重

**Reply by mark9804** [reply]
材料维持现状也行，把 boss 信息扔进 url 就可以了

fetch 的消耗是相对而言的，fetch 的总传输时间是 `请求进入传输队列等待的时间 + 服务器处理请求的时间 + 传输时间`，其中处理请求的时间一般是固定的，本地环境带宽足够大，所以理论上开 `n` 个小文件 `fetch` 所需要的时间就是 `n * （等待时间 + 处理时间）`

比如说 material.json 有 60k，fetch 需要 15ms；boss 信息就几行，但是 fetch 也要 12ms，就显得性价比很低

另外 material.json 很大，很多东西是不需要的，非常污染 url

https://github.com/Arondight/Adachi-BOT/blob/903603ef834cc6ce7492d774a901155cb3f8155a/src/views/genshin-material.html#L32-L36

**Comment by mark9804** [comment]
还有这个搞不好能抄

https://github.com/monsterxcn/nonebot-plugin-gsmaterial
